crates/brainfuck_vm/src/bin/brainfuck_vm.rs

```diff
@@ -38,7 +37,7 @@ fn main() {
     let stdout = stdout();
     let mut bf_vm = match args.ram_size {
         Some(size) => Machine::new_with_config(ins, stdin, stdout, size),
-        None => Machine::new(ins, stdin, stdout),
+        None => Machine::new(&ins, stdin, stdout),
```

Comment:
Why does `Machine::new()` signature takes a `&[BaseField]` for `code` (the compiled instructions) and `Machine::new_with_config` takes a `Vec<BaseField>`? I guess it should be uniform

Currently the compiler outputs a `Vec<BaseField>`, it should be of the same type as the Machine signature

Reply:
Yes you are right, this a clippy warning because clippy founds that it is more optimized to pass by ref rather than value so that I uniformized everything to `&[BaseField]` if this is ok for you

Reply:
Good for me, maybe also change the compiler output to `&[BaseField]` then? Can be done in another PR